src/Essentials/src/Platform/WindowStateManager.ios.cs

```diff
@@ -101,9 +101,18 @@ class WindowStateManagerImplementation : IWindowStateManager
 			// if we have scene support, use that
 			if (OperatingSystem.IsIOSVersionAtLeast(13) || OperatingSystem.IsMacCatalystVersionAtLeast(13))
 			{
-				var scenes = UIApplication.SharedApplication.ConnectedScenes;
-				var windowScene = scenes.ToArray<UIWindowScene>().FirstOrDefault();
-				return windowScene?.Windows.FirstOrDefault();
+				try
+				{
+					using var scenes = UIApplication.SharedApplication.ConnectedScenes;
+					var windowScene = scenes.ToArray<UIWindowScene>().FirstOrDefault();
+					return windowScene?.Windows.FirstOrDefault();
+				}
+				catch (InvalidCastException)
+				{
+					// HACK: Workaround for https://github.com/xamarin/xamarin-macios/issues/13704
+					//       This only throws if the collection is empty.
+					return null;
+				}
```

Comment:
Second level, if we do get an exception, we know it is because the set was empty, so do empty things.